README.md

```diff
@@ -118,13 +145,30 @@ VERSION=v1.0.0
 ENVIRONMENT=development
 ```
 
+## Templating
+
+Variables can by templated by using curly braces "`{{`" and "`}}`", in this example `envy-{{VERSION}}` is replaced with `envy-1.0.0`
+
+```
+VERSION=1.0.0
+NAME=envy-{{VERSION}}
+```
+
+To write a literal `{{`, escape it by putting it between curly braces `{{{{}}`
+
 ## Variable Precedence
 
-- Existing environment variables will not be overridden
-- Includes will override variables which are declared afterwards, if you have a shared common.env include add it to the bottom of your file so preceding variables can override its contents
+- Variables are processed top to bottom and lower varibles will overrite previous ones
+- Includes will be evaluated in the line they are written. If you have a shared common.env you can include it at the top of your file so following variables can override its values
+- Output will be sorted alphabetically at the end after the above evaluations for consistency
```

Comment:
Had to read that line point about sorting a couple of times, maybe reword it so it's clear that all lines generated by envy will be sorted alphabetically before the output is written. I think the 'after the above evaluations' throws it off a bit.

This and a nitpick at 'overrite' to 'overwrite'